src/pydata_sphinx_theme/__init__.py

```diff
@@ -51,6 +53,31 @@ def update_config(app, env):
             " Set version URLs in JSON directly."
         )
 
+    # check the validity of the theme swithcer file
+    if isinstance(theme_options.get("switcher"), dict):
+        theme_switcher = theme_options.get("switcher")
+        if theme_switcher.get("json_url"):
+            json_url = theme_switcher["json_url"]
+            theme_switcher["version_match"]
+
+            # try to read the json file. If it's a url we use request,
+            # else we simply read the local file from the source directory
+            try:
+                content = requests.get(json_url).text
+            except Exception:
+                content = Path(env.srcdir, json_url).read_text()
```

Comment:
if the file isn't there let's raise a helpful error message

Reply:
I changed the method, I parse the url if it's a https/http I try to read with request and pathlib if it's a local file. Both will raise an error if the file cannot be opened.